### ISSUE TITLE
Ubuntu 20.04: Errors on Install

### ISSUE THREAD

**Issue by remklov** [DidierRLopes/GamestonkTerminal#54]
I get several errors while installing the requirements on Ubuntu 20.04.

---

>   Building wheel for appscript (setup.py) ... error
ERROR: Command errored out with exit status 1:
   command: /usr/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-_47q6av7/appscript/setup.py'"'"'; __file__='"'"'/tmp/pip-install-_47q6av7/appscript/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-8zf7z1dv
       cwd: /tmp/pip-install-_47q6av7/appscript/
  Complete output (39 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-x86_64-3.8
  copying appscript_3x/lib/mactypes.py -> build/lib.linux-x86_64-3.8
  copying appscript_3x/lib/osax.py -> build/lib.linux-x86_64-3.8
  creating build/lib.linux-x86_64-3.8/aem
  copying appscript_3x/lib/aem/kae.py -> build/lib.linux-x86_64-3.8/aem
  copying appscript_3x/lib/aem/__init__.py -> build/lib.linux-x86_64-3.8/aem
  copying appscript_3x/lib/aem/aemcodecs.py -> build/lib.linux-x86_64-3.8/aem
  copying appscript_3x/lib/aem/aemconnect.py -> build/lib.linux-x86_64-3.8/aem
  copying appscript_3x/lib/aem/findapp.py -> build/lib.linux-x86_64-3.8/aem
  copying appscript_3x/lib/aem/aemsend.py -> build/lib.linux-x86_64-3.8/aem
  copying appscript_3x/lib/aem/typewrappers.py -> build/lib.linux-x86_64-3.8/aem
  copying appscript_3x/lib/aem/mactypes.py -> build/lib.linux-x86_64-3.8/aem
  copying appscript_3x/lib/aem/aemreference.py -> build/lib.linux-x86_64-3.8/aem
  creating build/lib.linux-x86_64-3.8/appscript
  copying appscript_3x/lib/appscript/terminology.py -> build/lib.linux-x86_64-3.8/appscript
  copying appscript_3x/lib/appscript/defaultterminology.py -> build/lib.linux-x86_64-3.8/appscript
  copying appscript_3x/lib/appscript/keywordwrapper.py -> build/lib.linux-x86_64-3.8/appscript
  copying appscript_3x/lib/appscript/__init__.py -> build/lib.linux-x86_64-3.8/appscript
  copying appscript_3x/lib/appscript/reservedkeywords.py -> build/lib.linux-x86_64-3.8/appscript
  copying appscript_3x/lib/appscript/referencerenderer.py -> build/lib.linux-x86_64-3.8/appscript
  copying appscript_3x/lib/appscript/genericreference.py -> build/lib.linux-x86_64-3.8/appscript
  copying appscript_3x/lib/appscript/reference.py -> build/lib.linux-x86_64-3.8/appscript
  copying appscript_3x/lib/appscript/terminologyparser.py -> build/lib.linux-x86_64-3.8/appscript
  running build_ext
  building 'aem.ae' extension
  creating build/temp.linux-x86_64-3.8
  creating build/temp.linux-x86_64-3.8/appscript_3x
  creating build/temp.linux-x86_64-3.8/appscript_3x/ext
  x86_64-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/usr/include/python3.8 -c appscript_3x/ext/ae.c -o build/temp.linux-x86_64-3.8/appscript_3x/ext/ae.o -DMAC_OS_X_VERSION_MIN_REQUIRED=MAC_OS_X_VERSION_10_6 -D__LP64__
  In file included from appscript_3x/ext/ae.c:16:
  appscript_3x/ext/ae.h:26:10: fatal error: Carbon/Carbon.h: Datei oder Verzeichnis nicht gefunden
     26 | #include <Carbon/Carbon.h>
        |          ^~~~~~~~~~~~~~~~~
  compilation terminated.
  error: command 'x86_64-linux-gnu-gcc' failed with exit status 1

---

> Building wheel for fbprophet (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /usr/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-_47q6av7/fbprophet/setup.py'"'"'; __file__='"'"'/tmp/pip-install-_47q6av7/fbprophet/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-o8c5hg2u
       cwd: /tmp/pip-install-_47q6av7/fbprophet/
  Complete output (44 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib
  creating build/lib/fbprophet
  creating build/lib/fbprophet/stan_model
  Traceback (most recent call last):
    File "<string>", line 1, in <module>
    File "/tmp/pip-install-_47q6av7/fbprophet/setup.py", line 122, in <module>
      setup(
    File "/usr/lib/python3/dist-packages/setuptools/__init__.py", line 144, in setup
      return distutils.core.setup(**attrs)
    File "/usr/lib/python3.8/distutils/core.py", line 148, in setup
      dist.run_commands()
    File "/usr/lib/python3.8/distutils/dist.py", line 966, in run_commands
      self.run_command(cmd)
    File "/usr/lib/python3.8/distutils/dist.py", line 985, in run_command
      cmd_obj.run()
    File "/usr/lib/python3/dist-packages/wheel/bdist_wheel.py", line 223, in run
      self.run_command('build')
    File "/usr/lib/python3.8/distutils/cmd.py", line 313, in run_command
      self.distribution.run_command(command)
    File "/usr/lib/python3.8/distutils/dist.py", line 985, in run_command
      cmd_obj.run()
    File "/usr/lib/python3.8/distutils/command/build.py", line 135, in run
      self.run_command(cmd_name)
    File "/usr/lib/python3.8/distutils/cmd.py", line 313, in run_command
      self.distribution.run_command(command)
    File "/usr/lib/python3.8/distutils/dist.py", line 985, in run_command
      cmd_obj.run()
    File "/tmp/pip-install-_47q6av7/fbprophet/setup.py", line 48, in run
      build_models(target_dir)
    File "/tmp/pip-install-_47q6av7/fbprophet/setup.py", line 36, in build_models
      from fbprophet.models import StanBackendEnum
    File "/tmp/pip-install-_47q6av7/fbprophet/fbprophet/__init__.py", line 8, in <module>
      from fbprophet.forecaster import Prophet
    File "/tmp/pip-install-_47q6av7/fbprophet/fbprophet/forecaster.py", line 17, in <module>
      from fbprophet.make_holidays import get_holiday_names, make_holidays_df
    File "/tmp/pip-install-_47q6av7/fbprophet/fbprophet/make_holidays.py", line 14, in <module>
      import fbprophet.hdays as hdays_part2
    File "/tmp/pip-install-_47q6av7/fbprophet/fbprophet/hdays.py", line 13, in <module>
      from convertdate.islamic import from_gregorian, to_gregorian
  ModuleNotFoundError: No module named 'convertdate'

---

> Running setup.py install for appscript ... error
    ERROR: Command errored out with exit status 1:
     command: /usr/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-_47q6av7/appscript/setup.py'"'"'; __file__='"'"'/tmp/pip-install-_47q6av7/appscript/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-j24seff8/install-record.txt --single-version-externally-managed --user --prefix= --compile --install-headers /home/[...]/.local/include/python3.8/appscript
         cwd: /tmp/pip-install-_47q6av7/appscript/
    Complete output (40 lines):
    running install
    Distribution option extra_path is deprecated. See issue27919 for details.
    running build
    running build_py
    creating build
    creating build/lib.linux-x86_64-3.8
    copying appscript_3x/lib/mactypes.py -> build/lib.linux-x86_64-3.8
    copying appscript_3x/lib/osax.py -> build/lib.linux-x86_64-3.8
    creating build/lib.linux-x86_64-3.8/aem
    copying appscript_3x/lib/aem/kae.py -> build/lib.linux-x86_64-3.8/aem
    copying appscript_3x/lib/aem/__init__.py -> build/lib.linux-x86_64-3.8/aem
    copying appscript_3x/lib/aem/aemcodecs.py -> build/lib.linux-x86_64-3.8/aem
    copying appscript_3x/lib/aem/aemconnect.py -> build/lib.linux-x86_64-3.8/aem
    copying appscript_3x/lib/aem/findapp.py -> build/lib.linux-x86_64-3.8/aem
    copying appscript_3x/lib/aem/aemsend.py -> build/lib.linux-x86_64-3.8/aem
    copying appscript_3x/lib/aem/typewrappers.py -> build/lib.linux-x86_64-3.8/aem
    copying appscript_3x/lib/aem/mactypes.py -> build/lib.linux-x86_64-3.8/aem
    copying appscript_3x/lib/aem/aemreference.py -> build/lib.linux-x86_64-3.8/aem
    creating build/lib.linux-x86_64-3.8/appscript
    copying appscript_3x/lib/appscript/terminology.py -> build/lib.linux-x86_64-3.8/appscript
    copying appscript_3x/lib/appscript/defaultterminology.py -> build/lib.linux-x86_64-3.8/appscript
    copying appscript_3x/lib/appscript/keywordwrapper.py -> build/lib.linux-x86_64-3.8/appscript
    copying appscript_3x/lib/appscript/__init__.py -> build/lib.linux-x86_64-3.8/appscript
    copying appscript_3x/lib/appscript/reservedkeywords.py -> build/lib.linux-x86_64-3.8/appscript
    copying appscript_3x/lib/appscript/referencerenderer.py -> build/lib.linux-x86_64-3.8/appscript
    copying appscript_3x/lib/appscript/genericreference.py -> build/lib.linux-x86_64-3.8/appscript
    copying appscript_3x/lib/appscript/reference.py -> build/lib.linux-x86_64-3.8/appscript
    copying appscript_3x/lib/appscript/terminologyparser.py -> build/lib.linux-x86_64-3.8/appscript
    running build_ext
    building 'aem.ae' extension
    creating build/temp.linux-x86_64-3.8
    creating build/temp.linux-x86_64-3.8/appscript_3x
    creating build/temp.linux-x86_64-3.8/appscript_3x/ext
    x86_64-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/usr/include/python3.8 -c appscript_3x/ext/ae.c -o build/temp.linux-x86_64-3.8/appscript_3x/ext/ae.o -DMAC_OS_X_VERSION_MIN_REQUIRED=MAC_OS_X_VERSION_10_6 -D__LP64__
    In file included from appscript_3x/ext/ae.c:16:
    appscript_3x/ext/ae.h:26:10: fatal error: Carbon/Carbon.h: Datei oder Verzeichnis nicht gefunden
       26 | #include <Carbon/Carbon.h>
          |          ^~~~~~~~~~~~~~~~~
    compilation terminated.
    error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
    ----------------------------------------
ERROR: Command errored out with exit status 1: /usr/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-_47q6av7/appscript/setup.py'"'"'; __file__='"'"'/tmp/pip-install-_47q6av7/appscript/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-j24seff8/install-record.txt --single-version-externally-managed --user --prefix= --compile --install-headers /home/[...]/.local/include/python3.8/appscript Check the logs for full command output.

Somebody had these issues or has an idea how to resolve?

**Comment by luger1** [comment]
This is some issue with appscript. Delete it from the requirements and install again. It shouldn't be necessary to run it, or so i heard. Grüße aus Köln :)

**Comment by foranpower** [comment]
This is probably obvious to you already, but it wasn't for me on linux.
-> from: requirement.txt
->delete: #//*not sure if these are necessary, but they were giving me errors in linux
-->xlwings 
-->appscript

cd the gamestonk terminal directory (or right click inside the folder and click open in terminal)
type in terminal: python gamestonk_terminal.py

Still, seems some menus are missing. Maybe xlwings was necessary.

**Comment by DidierRLopes** [comment]
Exactly correct @foranpower ,

The xlwings and appscript aren't necessary. A fixed and improved requirements.txt list is coming soon.

Regarding the "menus missing" that made me giggle :) It's one of those things where you try to do something fancy but it may not be that intuitive. The reason why the other menus don't appear is because you don't have a loaded stock ticker, meaning that you can only "discover new stocks", understand some "sentiment of the market".

If you do `load -t GME` you'll be able to see all the menu, and as soon as you go into one of these menus, the ticker GME will be pre-loaded. 

TL;DR: You're all set, enjoy!

**Comment by shadycuz** [comment]
I'm going to close this now. Lots has changed since this was opened. You can run `git checkout main`, `git pull main` to fetch the latest and attempt to reinstall.

The project has been testest to work with Pythons 3.6.8, 3.7 and 3.8. Tensorflow doesn't support python 3.9 yet.

We are using poetry to manage the project and that is one option for installing. The other is to install with the requirements.txt. Feel free to use any virtual environment you want.

Just make sure to run pip install --upgrade pip as that was affecting some people as tensorflow requires pip >= 2.19 (or something like that).

We also have a dockerfile if you would prefer to use docker. We will also soon be offering an "official image" on dockerhub.

Oh and Windows is only supported with docker because pystan is not available for windows (officially).

If you are having problems please open a new issue =).

🚀 GamestonkTerminal 🌔
